numba/tests/test_np_randomgen.py

```diff
@@ -138,6 +138,135 @@ def test_random(self):
+        bitgen_types = [None, MT19937]
+
+        dist_func = lambda x, size, dtype: \
+            x.standard_gamma(shape=5.0, size=size, dtype=dtype)
+        for _size in test_sizes:
+            for _dtype in test_dtypes:
+                for _bitgen in bitgen_types:
+                    with self.subTest(_size=_size, _dtype=_dtype,
+                                      _bitgen=_bitgen):
+                        self.check_numpy_parity(dist_func, _bitgen,
+                                                None, _size, _dtype)
+
+    def test_normal(self):
+        # For this test dtype argument is never used, so we pass [None] as dtype
+        # to make sure it runs only once with default system type.
+
+        test_sizes = [None, (), (100,), (10, 20, 30)]
+        bitgen_types = [None, MT19937]
+
+        # Test with no arguments
+        dist_func = lambda x, size, dtype:x.normal()
+        with self.subTest():
+            self.check_numpy_parity(dist_func, test_size=None,
+                                    test_dtype=None)
+
+        dist_func = lambda x, size, dtype:x.normal(loc=1.5, scale=3, size=size)
+        for _size in test_sizes:
+            for _bitgen in bitgen_types:
+                with self.subTest(_size=_size, _bitgen=_bitgen):
+                    self.check_numpy_parity(dist_func, _bitgen,
+                                            None, _size, None)
+
+    def test_exponential(self):
+        # For this test dtype argument is never used, so we pass [None] as dtype
+        # to make sure it runs only once with default system type.
+
+        test_sizes = [None, (), (100,), (10, 20, 30)]
+        bitgen_types = [None, MT19937]
+
+        # Test with no arguments
+        dist_func = lambda x, size, dtype:x.exponential()
+        with self.subTest():
+            self.check_numpy_parity(dist_func, test_size=None,
+                                    test_dtype=None)
+
+        dist_func = lambda x, size, dtype:x.exponential(scale=1.5, size=size)
+        for _size in test_sizes:
+            for _bitgen in bitgen_types:
+                with self.subTest(_size=_size, _bitgen=_bitgen):
+                    self.check_numpy_parity(dist_func, _bitgen,
+                                            None, _size, None)
+
+    def test_gamma(self):
+        # For this test dtype argument is never used, so we pass [None] as dtype
+        # to make sure it runs only once with default system type.
+
+        test_sizes = [None, (), (100,), (10, 20, 30)]
+        bitgen_types = [None, MT19937]
+
+        dist_func = lambda x, size, dtype:x.gamma(shape=5.0, scale=1.5,
+                                                  size=size)
+        for _size in test_sizes:
+            for _bitgen in bitgen_types:
+                with self.subTest(_size=_size, _bitgen=_bitgen):
+                    self.check_numpy_parity(dist_func, _bitgen,
+                                            None, _size, None)
+
```

Comment:
These tests are great for checking numerical behaviours. It would be good to see in addition tests that ensure invalid types are handled nicely. For example, this:
```python
@njit
def foo(rng):
    rng.exponential(scale=None)

foo(np.random.default_rng(1))
```
ends up with a really long traceback because you can't multiply `none` by a `float64`. Ideally the type of the variable `scale` would have been checked to make sure it's a `types.Float` or `types.Integer`. This applies to all unchecked variables in the `Generator.<method>(*args, **kwargs)` `@overload_method` definitions.

Reply:
@kc611 just pinging to make sure that this is noted as an unaddressed item from the previous review, all the other things look like they are covered. Thanks!